### DISCUSSION TITLE
bun:sqlite named parameters without special symbol

### DISCUSSION THREAD

**Discussion by vanodevium** [oven-sh/bun#11142] [Ideas]
### What is the problem this feature would solve?

In my opinion, this will greatly simplify and unify the use of named parameters, as well as allow in the future to create wrappers that will be able to use SQLite in different runtimes.

### What is the feature you are proposing to solve the problem?

Let's imagine we have next statement:
```js
const query = db.query("SELECT $foo, :bar, @baz");
```

For now we have to set parameters as they are described in the query, so we use this notation:
```js
const result = query.get({
      "$foo": 1,
      ":bar": 2,
      "@baz": 3,
});
```

It would be very useful to support also next notation:
```js
const result = query.get({
      foo: 1,
      bar: 2,
      baz: 3,
});
```

Is there any chance to implement this functionality?

**Comment by murataka** [comment]
better-sqlite3 library can do it without special symbols. when symbols are required, we have to destructure the input. However , destructuring should be left to developer's choice . Being forced to use symbols also makes the code a bit more ugly when reading.

**Reply by murataka** [reply]
oh just for people ending up here : 

`const db = new Database("./db.s3db", { strict: true });
`
fixes that
